### PR TITLE
Push 23.0.5 into main

### PR DESCRIPTION
**Pull request by ridz1208**
This merger came with some difficult conflicts to resolve, the following file changes should be reviewed thoroughly.

```
       deleted by us:   modules/behavioural_qc/templates/form_behavioural_qc.tpl
	both modified:   modules/configuration/templates/form_configuration.tpl
	both modified:   modules/document_repository/php/files.class.inc
	both modified:   modules/help_editor/help/help_editor.md
	both modified:   modules/help_editor/php/edit_help_content.class.inc
	both modified:   modules/media/ajax/FileUpload.php
	both modified:   modules/module_manager/jsx/modulemanager.js
	both modified:   php/libraries/NDB_BVL_Instrument.class.inc
	both modified:   php/libraries/User.class.inc
	both modified:   tools/exporters/data_dictionary_builder.php
```

### CODE REVIEW

File: modules/document_repository/php/files.class.inc [hunk 1]
```diff
@@ -317,7 +317,7 @@ class Files extends \NDB_Page
                 'No file uploaded'
             );
         }
-        $filename = $uploadedFile->getClientFilename();
+        $filename = urldecode($uploadedFile->getClientFilename());
```
Comment (ridz1208) [inline modules/document_repository/php/files.class.inc:320]: @driusan not sure if this is still necessary with the use of the file uploader class

Comment (ridz1208) [reply modules/document_repository/php/files.class.inc:320]: also not sure if the decode should happen IN the file uplodaer class

Comment (ridz1208) [reply modules/document_repository/php/files.class.inc:320]: 23: https://github.com/aces/Loris/pull/7383
main: https://github.com/aces/Loris/pull/5234

File: modules/document_repository/php/files.class.inc [hunk 2]
```diff
@@ -384,7 +383,7 @@ class Files extends \NDB_Page
         $fileType = pathinfo($filename, PATHINFO_EXTENSION);
 
         // Insert record of this file into the `document_repository` table.
-        $DB->insert(
+        $DB->unsafeInsertOnDuplicateUpdate(
```
Comment (ridz1208) [inline modules/document_repository/php/files.class.inc:386]: @driusan same PRs as above, I think this one is still necessary

File: modules/help_editor/php/edit_help_content.class.inc [hunk 3]
```diff
@@ -58,13 +58,21 @@ class Edit_Help_Content extends \NDB_Form
             $helpID = htmlspecialchars($_REQUEST['helpID']);
         }
         if (isset($_REQUEST['section']) && !empty($_REQUEST['section'])) {
-            $helpID = HelpFile::hashToID(md5($_REQUEST['section']));
+            try {
+                $helpID = HelpFile::hashToID(md5($_REQUEST['section']));
+            } catch (\NotFound $e) {
+                $helpID = '';
+            }
```
Comment (ridz1208) [inline modules/help_editor/php/edit_help_content.class.inc:65]: 23: https://github.com/aces/Loris/pull/6907/files#diff-ea2fe2cfe5a3b25014d5787a3984a861bece8432c699cb104608600f6f29274aR66
main: https://github.com/aces/Loris/pull/7025/files#diff-ea2fe2cfe5a3b25014d5787a3984a861bece8432c699cb104608600f6f29274aR63

File: modules/document_repository/php/files.class.inc [hunk 4]
```diff
@@ -343,7 +343,6 @@ class Files extends \NDB_Page
 
         $targetdir = new \SplFileInfo($path);
 
-        $filename = $uploadedFile->getClientFileName();
```
Comment (ridz1208) [inline modules/document_repository/php/files.class.inc:346]: deleted because its called on line 320

**Comment by kongtiaowang** [comment]
modify this modules/genomic_browser/test/genomic_browserTest.php
https://github.com/aces/Loris/pull/7543

**Comment by ridz1208** [comment]
Closed in favour of https://github.com/aces/Loris/pull/7544

**Pull request closed without merging**